### PR TITLE
Add `craft` command transition

### PR DESCRIPTION
**Pull request by bencroker**
## The Problem/Issue/Bug:

Once https://github.com/drud/ddev/pull/4176 is merged, existing Craft projects that have the project type set to `php` will no longer have the `craft` command available to them and will output an error.

Screenshot 2022-09-09 at 19 36 06 2

## How this PR Solves The Problem:

This PR maintains the `craft` command in the `php` project type, but prompts users to switch to the `craftcms` project type in order to use it, easing the transition for existing users who update DDEV without reading the release notes.

This is primarily to aid in transitioning and the `craft` command could be dropped for`php` project types in the next minor version, if desirable.

Screenshot 2022-09-09 at 19 36 06

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4199"></a>

### CODE REVIEW

**Comment by brandonkelly** [comment]
Would be great to have this, at least temporarily!

**Comment by khalwat** [comment]
While I realize this would still be polluting the `php` command namespace, we're doing that already... and this would ease the transition.

We could set a trigger to remove it in a year or something, once the transition should be mostly complete.

I will also add something to the docs to tell people how to manually change the project type.

**Comment by rfay** [comment]
Another option is just to let it work on php for a release or two, perhaps with a pester when you run it?

**Comment by bencroker** [comment]
> Another option is just to let it work on php for a release or two, perhaps with a pester when you run it?

My preference would be to force people to switch project types. Pestering often goes ignored and once we finally do remove it, there may be confusion for those that simply ignored the message.

**Comment by rfay** [comment]
Plesae rebase too, thanks.

**Comment by bencroker** [comment]
> Plesae rebase too, thanks.

Rebased and ready to merge.

File: pkg/ddevapp/global_dotddev_assets/commands/web/craft [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
-## ProjectTypes: craftcms
+## Example: "ddev craft db/backup" or "ddev craft db/backup ./my-backups" (see https://craftcms.com/docs/4.x/console-commands.html)
+## ProjectTypes: craftcms,php
 ## ExecRaw: true
 
-CRAFT_CMD_ROOT=${CRAFT_CMD_ROOT:="./"}
-
-cd "${CRAFT_CMD_ROOT}"
-php ./craft "$@"
+if grep -q "type: php" ./.ddev/config.yaml; then
```
Comment (rfay) [inline pkg/ddevapp/global_dotddev_assets/commands/web/craft:10]:
```suggestion
if [ "${DDEV_PROJECT_TYPE}" != "craftcms" ]; then
```

Comment (rfay) [reply pkg/ddevapp/global_dotddev_assets/commands/web/craft:10]: I didn't manually test this, so please give it a quick manual review, but think it's better.

File: pkg/ddevapp/global_dotddev_assets/commands/web/craft [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
+    echo "The craft command is only available in the craftcms project type. You can update this in your project's config file, followed by restarting the DDEV project."
+else
+    CRAFT_CMD_ROOT=${CRAFT_CMD_ROOT:="./"}
+    
+    cd "${CRAFT_CMD_ROOT}"
+    php ./craft "$@"
```
Comment (rfay) [inline pkg/ddevapp/global_dotddev_assets/commands/web/craft:16]: Just a note that `php ./craft` is pretty much redundant and `php craft` is the same. But that's not important. I do keep trying to understand if craft may be somewhere else, in which case we could consider using `## HostWorkingDir: true` to run it in the same relative directory they're in. I don't think there's any reason to fiddle with this right now, jut talking out loud.

**Review by rfay — CHANGES_REQUESTED** [review]
Thanks for this, and thanks for figuring out how to make things better for the Craft community.

File: pkg/ddevapp/global_dotddev_assets/commands/web/craft (hunk 2 above)
Comment (khalwat) [reply pkg/ddevapp/global_dotddev_assets/commands/web/craft:16]: Yeah the check for the environment variable is likely better, I agree. I just didn't know it existed.

Comment (bencroker) [reply pkg/ddevapp/global_dotddev_assets/commands/web/craft]: Updated, thanks for the review!

**Review by rfay — APPROVED** [review]
Thanks for this!

**Comment by rfay** [comment]
Manually tested this and it seems to be fine. Thanks!

**Pull request merged**